coreneuron/io/mech_report.cpp

```diff
@@ -0,0 +1,55 @@
+#include <iostream>
+#include <vector>
+
+#include "coreneuron/coreneuron.hpp"
+#include "coreneuron/mpi/nrnmpi.h"
+#include "coreneuron/mpi/nrnmpi_impl.h"
+
+namespace coreneuron {
+
+/** display global mechanism count */
+void write_mech_report() {
+    /// mechanim count across all gids, local to rank
+    const auto n_memb_func = corenrn.get_memb_funcs().size();
+    std::vector<unsigned long long> local_mech_count(n_memb_func, 0);
+
+    /// each gid record goes on separate row, only check non-empty threads
+    for (size_t i = 0; i < nrn_nthread; i++) {
+        const auto& nt = nrn_threads[i];
+        for (auto* tml = nt.tml; tml; tml = tml->next) {
+            const int type = tml->index;
+            const auto& ml = tml->ml;
+            local_mech_count[type] += ml->nodecount;
+        }
+    }
+
+#if NRNMPI
+    /// get global sum of all mechanism instances
+    std::vector<unsigned long long> total_mech_count(n_memb_func);
```

Comment:
nitpick / suggestion : initialise `total_mech_count = local_mech_count` outside `#if NRNMPI` and then have below printing code common. (`if (nrnmpi_myid == 0)` can be also common because the `nrnmpi_myid` variable exist with 0).

Reply:
ah, the latter I was wondering about - in this case I can do that